### PR TITLE
New UnsafeDeserialization sink and improvements to SnakeYaml sink

### PR DESCRIPTION
**Pull request by pwntester**

### CODE REVIEW

File: java/ql/src/semmle/code/java/frameworks/SnakeYaml.qll [hunk 1]
```diff
@@ -71,7 +71,7 @@ private class SnakeYamlParse extends MethodAccess {
   SnakeYamlParse() {
     exists(Method m |
       m.getDeclaringType() instanceof Yaml and
-      (m.hasName("load") or m.hasName("loadAll") or m.hasName("loadAs") or m.hasName("parse")) and
+      (m.hasName("compose") or m.hasName("composeAll") or m.hasName("load") or m.hasName("loadAll") or m.hasName("loadAs") or m.hasName("parse")) and
```
Comment (aschackmull) [inline java/ql/src/semmle/code/java/frameworks/SnakeYaml.qll:74]: We can simplify a bit:
```suggestion
      m.hasName(["compose", "composeAll", "load", "loadAll", "loadAs", "parse"]) and
```

**Comment by aschackmull** [comment]
Looks good. Could you add a change note stating that the unsafe deserialization query has been improved with the additional sink? Look in `java/change-notes` for examples.

File: java/change-notes/2020-11-04-commonslang-unsafe-deserialization-sinks.md [hunk 2]
```diff
@@ -0,0 +1,4 @@
+lgtm,codescanning
+* The Unsafe Deserialization queries have been improved to recognize unsafe Apache Commons Lang(3)? methods.
```
Comment (aschackmull) [inline java/change-notes/2020-11-04-commonslang-unsafe-deserialization-sinks.md:2]: There is just one, right? Also, the question mark seems questionable.
```suggestion
* The query "Deserialization of user-controlled data" (`java/unsafe-deserialization`) has been improved to recognize unsafe Apache Commons Lang(3) methods.
```

File: java/ql/src/semmle/code/java/frameworks/SnakeYaml.qll (hunk 1 above)
Comment (pwntester) [reply java/ql/src/semmle/code/java/frameworks/SnakeYaml.qll]: Cool, knew about `m.getName() in [...]` but not that you could pass an array to `hasName`

**Review by aschackmull — DISMISSED** [review]

File: java/ql/src/semmle/code/java/frameworks/SnakeYaml.qll (hunk 1 above)
Comment (aschackmull) [reply java/ql/src/semmle/code/java/frameworks/SnakeYaml.qll]: It's not an array - it's just multiple values. It's a general language feature that works everywhere. It means the same as introducing a new variable and giving that variable multiple possible values using `or`.

**Comment by aschackmull** [comment]
```
Error: The following CodeQL elements are lacking documentation:
semmle/code/java/frameworks/apache/Lang.qll  Lang::MethodApacheSerializationUtilsDeserialize  class
```

File: java/ql/src/semmle/code/java/frameworks/apache/Lang.qll [hunk 3]
```diff
@@ -10,3 +10,13 @@ class TypeApacheRandomStringUtils extends Class {
     hasQualifiedName("org.apache.commons.lang3", "RandomStringUtils")
   }
 }
+
+class MethodApacheSerializationUtilsDeserialize extends Method {
```
Comment (aschackmull) [inline java/ql/src/semmle/code/java/frameworks/apache/Lang.qll:14]: As the error indicates, this class needs qldoc.

File: java/ql/src/semmle/code/java/frameworks/apache/Lang.qll [hunk 4]
```diff
@@ -10,3 +10,17 @@ class TypeApacheRandomStringUtils extends Class {
     hasQualifiedName("org.apache.commons.lang3", "RandomStringUtils")
   }
 }
+
+/*--- Methods ---*/
+/** The method `deserialize` in either `org.apache.commons.lang.SerializationUtils`
```
Comment (aschackmull) [inline java/ql/src/semmle/code/java/frameworks/apache/Lang.qll:15]: This qldoc is not formatted correctly, and indeed we get:
```
[ERROR] Input file ql/java/ql/src/semmle/code/java/frameworks/apache/Lang.qll is not correctly formatted
```

**Review by aschackmull — APPROVED** [review]

**Pull request merged**